force-app/main/default/lwc/lookup/lookup.js

```diff
@@ -0,0 +1,422 @@
+        this._searchTerm = newSearchTerm;
+
+        // Compare clean new search term with current one and abort if identical
+        const newCleanSearchTerm = newSearchTerm.trim().replace(REGEX_SOSL_RESERVED, '?').toLowerCase();
+        if (this._cleanSearchTerm === newCleanSearchTerm) {
+            return;
+        }
+
+        // Save clean search term
+        this._cleanSearchTerm = newCleanSearchTerm;
+
+        // Ignore search terms that are too small after removing special characters
+        if (newCleanSearchTerm.replace(/\?/g, '').length < this.minSearchTermLength) {
+            this.setSearchResults(this._defaultSearchResults);
+            return;
+        }
+
+        // Apply search throttling (prevents search if user is still typing)
+        if (this._searchThrottlingTimeout) {
+            clearTimeout(this._searchThrottlingTimeout);
+        }
+        // eslint-disable-next-line @lwc/lwc/no-async-operation
+        this._searchThrottlingTimeout = setTimeout(() => {
+            // Send search event if search term is long enougth
+            if (this._cleanSearchTerm.length >= this.minSearchTermLength) {
+                // Display spinner until results are returned
+                this.loading = true;
+
+                const searchEvent = new CustomEvent('search', {
+                    detail: {
+                        searchTerm: this._cleanSearchTerm,
+                        rawSearchTerm: newSearchTerm,
+                        selectedIds: this._curSelection.map((element) => element.id)
+                    }
+                });
+                this.dispatchEvent(searchEvent);
+            }
+            this._searchThrottlingTimeout = null;
+        }, SEARCH_DELAY);
+    }
+
+    isSelectionAllowed() {
+        if (this.isMultiEntry) {
+            return true;
+        }
+        return !this.hasSelection();
+    }
+
+    hasSelection() {
+        return this._curSelection.length > 0;
+    }
+
+    processSelectionUpdate(isUserInteraction) {
+        // Reset search
+        this._cleanSearchTerm = '';
+        this._searchTerm = '';
+        this.setSearchResults([...this._defaultSearchResults]);
+        // Indicate that component was interacted with
+        this._isDirty = isUserInteraction;
+        // If selection was changed by user, notify parent components
+        if (isUserInteraction) {
+            const selectedIds = this._curSelection.map((sel) => sel.id);
+            this.dispatchEvent(new CustomEvent('selectionchange', { detail: selectedIds }));
+        }
+    }
+
+    // EVENT HANDLING
+
+    handleInput(event) {
+        // Prevent action if selection is not allowed
+        if (!this.isSelectionAllowed()) {
+            return;
+        }
+        this.updateSearchTerm(event.target.value);
+    }
+
+    handleKeyDown(event) {
+        if (this._focusedResultIndex === null) {
+            this._focusedResultIndex = -1;
+        }
+        if (event.keyCode === KEY_ARROW_DOWN) {
```

Comment:
Heads up that `keyCode` is deprecated:

https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode

It seems the recommendation is to use `KeyboardEvent.code` instead:

https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code

Though be aware that IE doesn't support it

https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code#browser_compatibility

If that is okay, here is the list of the `KeyboardEvent.key` constants

https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values

Reply:
SF classic still supports IE11, so this will stay for now